drivers/sdcard_spi/sdcard_spi.c

```diff
@@ -0,0 +1,1039 @@
+            return r1;
+        }
+        tried++;
+    } while ((max_retries < 0) || (tried <= max_retries));
+
+    DEBUG("_wait_for_r1: [TIMEOUT]\n");
+    return r1;
+}
+
+void _select_card_spi(sd_card_t *card)
+{
+    spi_acquire(card->spi_dev);
+    gpio_clear(card->cs_pin);
+}
+
+void _unselect_card_spi(sd_card_t *card)
+{
+    gpio_set(card->cs_pin);
+    spi_release(card->spi_dev);
+}
+
+static inline int _sw_spi_transfer_byte(sd_card_t *card, char out, char *in){
+    char rx = 0;
+    int i = 7;
+    for(; i >= 0; i--){
+        if( ((out >> (i)) & 0x01) == 1){
+            gpio_set(card->mosi_pin);
+        }else{
+            gpio_clear(card->mosi_pin);
+        }
+        xtimer_usleep(SD_CARD_PREINIT_CLOCK_PERIOD_US/2);
+        gpio_set(card->clk_pin);
+        rx = (rx | (gpio_read(card->miso_pin) << i));
+        xtimer_usleep(SD_CARD_PREINIT_CLOCK_PERIOD_US/2);
+        gpio_clear(card->clk_pin);
+    }
+    *in = rx;
+    return 1;
+}
+
+static inline int _hw_spi_transfer_byte(sd_card_t *card, char out, char *in){
+    return spi_transfer_byte(card->spi_dev, out, in);
+}
+
+static inline int _transfer_bytes(sd_card_t *card, char *out, char *in, unsigned int length){
+    int trans_ret;
+    unsigned trans_bytes = 0;
+    char in_temp;
+
+    for (trans_bytes = 0; trans_bytes < length; trans_bytes++) {
+        if (out != NULL) {
+            trans_ret = _dyn_spi_transfer_byte(card, out[trans_bytes], &in_temp);
+        }
+        else {
+            trans_ret = _dyn_spi_transfer_byte(card, SD_CARD_DUMMY_BYTE, &in_temp);
+        }
+        if (trans_ret < 0) {
+            return -1;
+        }
+        if (in != NULL) {
+            in[trans_bytes] = in_temp;
+        }
+    }
+
+    return trans_bytes;
+}
+
+static sd_rw_response_t _read_data_packet(sd_card_t *card, char token, char *data, int size)
+{
+    DEBUG("_read_data_packet: size: %d\n", size);
+    if (_wait_for_token(card, token, SD_DATA_TOKEN_RETRY_CNT) == true) {
+        DEBUG("_read_data_packet: [GOT TOKEN]\n");
+    }
+    else {
+        DEBUG("_read_data_packet: [GOT NO TOKEN]\n");
+        return SD_RW_NO_TOKEN;
+    }
+
+    if (_transfer_bytes(card, NULL, data, size) == size) {
+
+        DEBUG("_read_data_packet: data: ");
+        for (int i = 0; i < size; i++) {
+            DEBUG("0x%02X ", data[i]);
+        }
+        DEBUG("\n");
+
+        char crc_bytes[2];
+        if (_transfer_bytes(card, 0, crc_bytes, sizeof(crc_bytes)) == sizeof(crc_bytes)) {
+            uint16_t data__crc_16 = (crc_bytes[0] << 8) | crc_bytes[1];
+
+            if (_crc_16(data, size) == data__crc_16) {
+                DEBUG("_read_data_packet: [OK]\n");
+                return SD_RW_OK;
+            }
+            else {
+                DEBUG("_read_data_packet: [CRC_MISMATCH]\n");
+                return SD_RW_CRC_MISMATCH;
+            }
+
+        }
+        else {
+            DEBUG("_read_data_packet: _transfer_bytes [RX_TX_ERROR] (while transmitting crc)\n");
```

Comment:
same error message as below, remove here completely and remove `else` below, so it will hit the message in both cases?